### DISCUSSION TITLE
Cloned repositories breaking repeatedly

### DISCUSSION THREAD

**Discussion by pillai-s** [community/community#68231] [New to GitHub] [answered]
I am facing problems with git repositories. 
The repositories seem to be broken every time I try to do a new commit or pull from master. This happens with all repositories and the only way I was able to fix it (temporarily) is reclone every thing from scratch. The problem disappears momentarily and appears again after a few days ( this is inconsistent and I am unable to figure if there is a pattern).

The following message appears when this problem occurs:
```
fatal: unable to read tree e036bf0d757a3ad64619f2b9e5a95d8977ede6a2
```
I have looked for some assistance online but it doesn't help:
https://stackoverflow.com/questions/1507463/how-to-deal-with-git-could-not-read-error

I am using a windows system.
For pulling or pushing, I have used github desktop, git kraken, visual studio code, and windows terminal, all giving the same result.

Kindly assist.

**Comment by seyLu** [comment]
#### Hi @pillai-s,

Are you by chance using a dated windows machine? If so, you might want to replace your hard disk soon.

You might also want to check if there are any corrupted objects in git:
```bash
git fsck  # file system check

# if there are any, then run
git gc  # garbage collection
```

You can read more on:
- [Git docs: `git fsck`](https://git-scm.com/docs/git-fsck)
- [Git docs: `git gc`](https://git-scm.com/docs/git-gc)

**Reply by pillai-s** [reply]
Hello @seyLu,

Thank you for responding. My system is almost brand new, so I do not think it is a problem of the hard drive.

I ran the commands you suggested and i get a long list of lines starting with "missing blob".

Further, running git gc shows the following:

```
error: Could not read b6a225e7957c03b104268ce297ab2a8b904ef154
error: Could not read b6a225e7957c03b104268ce297ab2a8b904ef154
error: Could not read b6a225e7957c03b104268ce297ab2a8b904ef154
error: Could not read b6a225e7957c03b104268ce297ab2a8b904ef154
warning: reflog of 'HEAD' references pruned commits
warning: reflog of 'refs/heads/master' references pruned commits
warning: reflog of 'refs/remotes/origin/HEAD' references pruned commits
warning: reflog of 'refs/remotes/origin/master' references pruned commits
error: Could not read b6a225e7957c03b104268ce297ab2a8b904ef154
fatal: Failed to traverse parents of commit 0e3afbcadf2c0071ec2c60accc90a79b0732c42a
fatal: failed to run repack
```

**Comment by milindaShehan** [comment]
**Check Disk Health:**

- On Windows, open the Command Prompt with administrative privileges. You can do this by searching for "cmd" or "Command Prompt" in the Start menu, then right-click and choose "Run as administrator."
- Type the following command and press Enter: ` chkdsk /f`
- If prompted, confirm that you want to schedule a check for the next time you restart your computer.
- Restart your computer. The disk check will run before Windows starts.

**File System Check:**

- Open the Command Prompt with administrative privileges (as explained above).
- Type the following command and press Enter:  `sfc /scannow`
- Let the system scan and repair any corrupted system files.

**Disable Anti-Virus/Security Software:**

- Locate your anti-virus or security software in the system tray (usually at the bottom-right corner of your screen).
- Right-click on its icon and look for an option like "Disable," "Pause," or "Suspend Protection."
- Disable the software temporarily.

**Reinstall Git:**

- Go to the Control Panel on your computer.
- Click on "Programs" or "Programs and Features."
- Locate Git in the list of installed programs, click on it, and select "Uninstall."
- Download the latest version of Git from the official website and install it.

**Git Garbage Collection:**

- Open a Command Prompt.
- Navigate to your repository's directory using the cd command (e.g.,  `cd C:\path\to\your\repository` ).
- Run the command:  `git gc --prune=now`

**Check for Disk Space:**

- Open File Explorer and go to the drive where your Git repositories are stored.
- Check the available free space.

**Check for Repository Corruption:**

- Open a Command Prompt and navigate to your repository's directory.
- Run the command:  `git fsck`
- Follow any instructions it provides to fix any issues.

**Update Git Configuration:**

Open a Command Prompt and run the following commands:

> 
> `git config --global core.autocrlf true`
> 
> `git config --global core.filemode false`

**Check for Proxy/VPN Issues:**

- If you're using a proxy or VPN, ensure they are configured correctly or try disabling them temporarily to see if it resolves the issue.

**Reply by pillai-s** [reply]
Hello @milindaShehan. Thank you for responding. I tried all the things you suggested. and at the end when i check the status of the repo, i still get the fatal error.

**Comment by pillai-s** [comment, marked as answer]
The repo which breaks was one which was used to maintain CAD files designed in `PTC Creo`. `PTC Creo` handles different file revision by incrementing a counter at the end of the file name, after the file extension (e.g. myPart.prt.1 or myAssembly.asm.1).

Usually, when saving `PTC Creo` creates a new file with the same name and extension, but with the version incremented of 1 unit. While this behavior is desiderable to easily retrive previous version of files, is not suitable to store the data in a version control system.

`Spekan Purge Tool` was used to easily purge all the files that have a revision smaller than the highest, and rename the highest to ".1". The problem of the repos breaking was due to clicking on the `DELETE` button of the `Spekan Purge Tool`. The `DELETE` button removes the `.idx` files that `PTC Creo` generates from the directories its pointed at, and in the process also removes the `.idx` files from the `.git` folder which  identifies the folder as a GitHub repository. 

Therefore, to prevent the repos from breaking, the `DELETE` button on the `Spekan Purge Tool` should not be clicked.

Closing the issue as solved.
